lib/text_parser.dart

```diff
@@ -958,6 +963,97 @@ class TextParser extends StatelessWidget {
       child: widget,
     );
   }
+
+  Widget _addWidgetSpanToWidget({required Widget widgetSpan, required Widget nodeParsed}) {
+    if (nodeParsed is CleanRichText) {
+      return _addWidgetSpanToCleanRichText(widgetSpan: widgetSpan, richText: nodeParsed);
+    } if (nodeParsed is Column && nodeParsed.children.isNotEmpty) {
+      return _addWidgetSpanToColumn(widgetSpan: widgetSpan, column: nodeParsed);
+    } else {
+      return CleanRichText(
+        _optimizeTextspan(TextSpan(
+          children: [
+            WidgetSpan(child: nodeParsed),
+            WidgetSpan(child: widgetSpan)
+          ]
+        )),
+        maxLines: maxLines
+      );
+    }
+  }
+
+  Widget _addWidgetSpanToCleanRichText({required Widget widgetSpan, required CleanRichText richText}) {
+    final childRichText = richText.child;
+    if (childRichText is TextSpan) {
+      return CleanRichText(
+        _optimizeTextspan(TextSpan(
+          text: childRichText.text,
+          children: [
+            if (childRichText.children?.isNotEmpty == true)
```

Comment:
`== true` shouldn't be needed. Same below

Reply:
I do not think that. Can you explain why it's not needed?

Reply:
<img width="631" alt="Screenshot 2023-05-22 at 13 11 52" src="https://github.com/Sorunome/flutter_matrix_html/assets/80730648/bc71b540-906b-412c-a0a2-b746e5ce87a1">

The `Children` attribute is optional in TextSpan so IMO it can be null or empty